packages/server/src/models/EventModel.ts

```diff
@@ -1,9 +1,22 @@
 import { Event, EventType } from '../services/database/types';
 import BaseModel, { UuidType } from './BaseModel';
+import { Day } from '../utils/time';
+import { DbConnection } from '../db';
+import { NewModelFactoryHandler } from './factory';
+import { Config } from '../utils/types';
 
 
 export default class EventModel extends BaseModel<Event> {
 
+	private readonly eventTtl_: number;
+	private readonly eventAutoDelete_: boolean;
+
+	public constructor(db: DbConnection, dbSlave: DbConnection, modelFactory: NewModelFactoryHandler, config: Config) {
+		super(db, dbSlave, modelFactory, config);
+		this.eventAutoDelete_ = config.EVENTS_AUTO_DELETE_ENABLED;
+		this.eventTtl_ = config.EVENTS_AUTO_DELETE_AFTER_DAYS * Day;
+	}
```

Comment:
Actually the config is available from TaskService, so could you please do the following:

- Remove the eventTtl_ and eventAutoDelete_ from here
- Make deleteOldEvents take the TTL as argument
- In TaskService, only enable the task if EVENTS_AUTO_DELETE_ENABLED is true (same as USER_DATA_AUTO_DELETE_ENABLED)
- Still from TaskService pass EVENTS_AUTO_DELETE_AFTER_DAYS * Day as argument

This is to make the way it works more explicit and to avoid having to modify a global object (the config) in tests, which can have unexpected results.

Reply:
> In TaskService, only enable the task if 
That was already the case.

I made the changes you requested but had to remove a few tests that no longer made sense.